pkg/utils/gardener/shoot.go

```diff
@@ -784,9 +784,15 @@ func ComputeTechnicalID(projectName string, shoot *gardencorev1beta1.Shoot) stri
 	return fmt.Sprintf("%s-%s--%s", v1beta1constants.TechnicalIDPrefix, projectName, shoot.Name)
 }
 
-// IsShootNamespace returns true if the given namespace is a shoot namespace, i.e. it starts with the technical id prefix.
-func IsShootNamespace(namespace string) bool {
-	return strings.HasPrefix(namespace, v1beta1constants.TechnicalIDPrefix)
```

Comment:
Couldn't we keep this more simple (no context/client/error handling) if we just also checked for the `kube-system` namespace? I think we might have done this already in other places.

Reply:
Hmm, I considered this, too.
While it might work for the current usages, we might stumble upon this when running some controllers against `kube-system` of the shoot, e.g., something like adding the networkpolicy controller to resource-manager for the shoot's kube-system namespace.
I wanted to avoid a trap like this 😄

Reply:
Fair enough, we can keep it like this - it's "more correct" anyways and not significantly more complex. Thanks!